runtime/src/accounts_db.rs

```diff
@@ -701,6 +701,74 @@ impl AccountsDB {
         accounts_index.uncleaned_roots.clear();
     }
 
+    fn inc_store_counts(
+        id: AppendVecId,
+        purges: &HashMap<Pubkey, Vec<(Slot, AccountInfo)>>,
+        store_counts: &mut HashMap<AppendVecId, usize>,
+        already_counted: &mut HashSet<AppendVecId>,
+    ) {
+        if already_counted.contains(&id) {
+            return;
+        }
+        *store_counts.get_mut(&id).unwrap() += 1;
+        let mut keys = HashSet::new();
+        for (key, account_infos) in purges {
+            for (_slot, account_info) in account_infos {
+                if account_info.store_id == id {
+                    keys.insert(key);
+                }
+            }
+        }
+        already_counted.insert(id);
```

Comment:
nits: This could be moved to [its related code here](https://github.com/solana-labs/solana/pull/9447/files#diff-2099c5256db4eb5975c8834af38f6456R7) immediately after the `if` or create `else`.